### PR TITLE
doc2docx handler.

### PR DESCRIPTION
**Pull request by bbfsdev**

### CODE REVIEW

File: app/doc2html/conversionFunctions.py [hunk 1]
```diff
@@ -18,37 +23,49 @@
...
     try:
-        stdout, stderr = p.communicate(timeout=15)
-    except TimeoutError:
-        p.kill()
+        with simpleflock.SimpleFlock(soffice_lock_file, timeout=60):
+            logger.debug('running: %s' % cmd)
+            p = subprocess.Popen(cmd, stdout=subprocess.PIPE,
+                                 stderr=subprocess.PIPE, shell=True)
+            try:
+                stdout, stderr = p.communicate(timeout=30)
```
Comment (edoshor) [inline app/doc2html/conversionFunctions.py]: Can we make `timeout`a function of the number of files ?
Something like `min(60, max(10, number of files))`

File: app/doc2html/conversionFunctions.py [hunk 2]
```diff
@@ -18,37 +23,49 @@
...
+            except TimeoutError:
+                p.kill()
+                raise
+            else:
+                logger.debug(stdout.decode(encoding='UTF-8'))
+                if len(stderr) > 0:
+                    raise IOError(stderr)
+    except:
+        logger.debug("Failed or timedout when trying to lock soffice.")
```
Comment (edoshor) [inline app/doc2html/conversionFunctions.py]: Make it error level log entry
This message is a bit misleading as this except block is handling more then just locking problems.

File: app/doc2html/views.py [hunk 3]
```diff
@@ -1,65 +1,187 @@
...
+    # List of required conversions from doc to docx.
+    doc_to_docx_list = [None]*len(uids)
+    ret = [None]*len(uids)
+
+    # Download files to one directory (we need this for docx conversion).
+    # Then convers them in this one directory and copy to their own directory.
+    # TODO: Consider making this in parallel.
+    for idx, uid in enumerate(uids):
+        # Get real file type from mdb.
+        file_type = get_file_type(uid)
```
Comment (edoshor) [inline app/doc2html/views.py]: This is a classic N+1 problem. please call the DB only once, you call it in a loop...

File: app/doc2html/views.py [hunk 4]
```diff
@@ -1,65 +1,187 @@
...
+        # Download file if necessary.
+        if not os.path.exists(path):
+            move_list[idx].append(Move(path, dest_path))
+
+            url = current_app.config['LINKER_URL'] + uid
+            # TODO: Handle http errors and map them to relevant errors.
+            try:
+                request.urlretrieve(url, path)
+            except:
+                current_app.logger.debug('Failed urlretrieve for %s' % url)
```
Comment (edoshor) [inline app/doc2html/views.py]: `urlretrieve` is not so meaningful, try something in English

File: app/doc2html/views.py [hunk 5]
```diff
@@ -1,65 +1,187 @@
...
+                return ret
```
Comment (edoshor) [inline app/doc2html/views.py]: one error kills entire batch ?

File: app/doc2html/views.py [hunk 6]
```diff
@@ -1,65 +1,187 @@
...
+
+# Download files to one directory.
+# Converts doc to docx if needed.
+# Moves files to their directories.
+# Returns the destination docx paths.
+def process_docx_uid(uids):
+    if not len(uids):
+        return []
+
+    uid_first_dir = get_and_create_dir(uids[0])
```
Comment (edoshor) [inline app/doc2html/views.py]: use a temp dir.

**Review by edoshor — CHANGES_REQUESTED** [review]

File: app/doc2html/conversionFunctions.py (hunk 1 above)
Comment (bbfsdev) [reply app/doc2html/conversionFunctions.py]: Done.

File: app/doc2html/conversionFunctions.py (hunk 2 above)
Comment (bbfsdev) [reply app/doc2html/conversionFunctions.py]: Timeout is not an error, it is 503. Warning.

**Pull request closed without merging**